entrypoint.sh

```diff
@@ -1,46 +1,55 @@
-#!/bin/sh
+#!/usr/bin/dumb-init /bin/sh
 
-touch ${SSH_KEY_FILE:=/id_rsa}
-chmod 0400 ${SSH_KEY_FILE:=/id_rsa}
+# Set up key file
+KEY_FILE=${SSH_KEY_FILE:=/id_rsa}
+if [ ! -f "${KEY_FILE}" ]; then
+    echo "[ERROR] No SSH Key file found"
+    exit 1
+fi
+eval $(ssh-agent -s)
+cat "${SSH_KEY_FILE}" | ssh-add -k -
 
+# Set up known_hosts file if needed
 STRICT_HOSTS_KEY_CHECKING=no
 KNOWN_HOSTS=${SSH_KNOWN_HOSTS:=/known_hosts}
 if [ -f "${KNOWN_HOSTS}" ]; then
-    chmod 0400 ${KNOWN_HOSTS}
-    KNOWN_HOSTS_ARG="-o UserKnownHostsFile=${KNOWN_HOSTS}"
+    KNOWN_HOSTS_ARG="-o UserKnownHostsFile=${KNOWN_HOSTS} "
+    if [ -z "${SSH_STRICT_HOST_IP_CHECK:=}" ]; then
+        KNOWN_HOSTS_ARG="${KNOWN_HOSTS_ARG}-o CheckHostIP=no "
+    fi
     STRICT_HOSTS_KEY_CHECKING=yes
 fi
 
 # Pick a random port above 32768
 DEFAULT_PORT=$RANDOM
 let "DEFAULT_PORT += 32768"
-echo [INFO] Tunneling ${SSH_HOSTUSER:=root}@${SSH_HOSTNAME:=localhost}:${SSH_TUNNEL_REMOTE:=${DEFAULT_PORT}} to ${SSH_TUNNEL_HOST=localhost}:${SSH_TUNNEL_LOCAL:=22}
-eval $(ssh-agent -s)
-cat ${SSH_KEY_FILE} | ssh-add -k -
-echo autossh \
- -M 0 \
- -N \
- -o StrictHostKeyChecking=${STRICT_HOSTS_KEY_CHECKING} ${KNOWN_HOSTS_ARG:=} \
- -o ServerAliveInterval=10 \
- -o ServerAliveCountMax=3 \
- -o ExitOnForwardFailure=yes \
- -t -t \
- ${SSH_MODE:=-R} ${SSH_TUNNEL_REMOTE}:${SSH_TUNNEL_HOST}:${SSH_TUNNEL_LOCAL} \
- -p ${SSH_HOSTPORT:=22} \
- ${SSH_HOSTUSER}@${SSH_HOSTNAME}
 
+# Determine command line flags
+COMMAND="autossh "\
+"-M 0 "\
+"-N "\
+"-o StrictHostKeyChecking=${STRICT_HOSTS_KEY_CHECKING} ${KNOWN_HOSTS_ARG:=}"\
+"-o ServerAliveInterval=10 "\
+"-o ServerAliveCountMax=3 "\
+"-o ExitOnForwardFailure=yes "\
+"-t -t "\
+"${SSH_MODE:=-R} ${SSH_TUNNEL_REMOTE}:${SSH_TUNNEL_HOST}:${SSH_TUNNEL_LOCAL} "\
+"-p ${SSH_HOSTPORT:=22} "\
+"${SSH_HOSTUSER}@${SSH_HOSTNAME}"
+
+# Log to stdout
+INFO_TUNNEL_SRC="${SSH_HOSTUSER:=root}@${SSH_HOSTNAME:=localhost}:${SSH_TUNNEL_REMOTE:=${DEFAULT_PORT}}"
+INFO_TUNNEL_DEST="${SSH_TUNNEL_HOST=localhost}:${SSH_TUNNEL_LOCAL:=22}"
```

Comment:
You are defaulting variables (`SSH_HOSTUSER`, `SSH_HOSTNAME`, etc) below, but using them un-defaulted above.

Please move the `INFO` statements above the `COMMAND` statement.

If you run it without variables, you get the following, which is probably not what you want.

```
[INFO] Tunneling root@localhost:44782 to localhost:22
> autossh -M 0 -N -o StrictHostKeyChecking= -o ServerAliveInterval=10 -o ServerAliveCountMax=3
     -o ExitOnForwardFailure=yes -t -t -R :: -p 22 @
```